.gitattributes

```diff
@@ -1,3 +1,10 @@
+# Checkout as native, commit as LF except in specific circumstances
+* text=auto
+*.bat text eol=crlf
+*.rc text eol=crlf
+*.sln text eol=crlf
```

Comment:
Do we need `.sln`? There is only one file in `clang/tools/clang-format-vs`. There are a few other text files in this directory, so perhaps a gitattributes in that directory is better?

Reply:
I think it makes sense to put the well-known-extension attributes in the root. I don't envision a `.sln` file ever *not* having crlf endings, so I considered it a global convenience